node/src/main/scala/com/wavesplatform/state/BlockchainUpdaterImpl.scala

```diff
@@ -124,6 +124,45 @@ class BlockchainUpdaterImpl(private val blockchain: LevelDBWriter,
     }
   }
 
+  private def rewardForBlock(block: Block): Option[Long] = {
+    val settings = blockchain.settings.rewardsSettings
+    val height   = blockchain.height + 1
+
+    blockchain.featureActivationHeight(BlockchainFeatures.BlockReward.id)
+      .filter(_ <= height)
+      .flatMap { activatedAt =>
+        def votes: Seq[Long] = {
+          val votes       = blockchain.blockRewardVotes(height - 1)
+          val currentVote = block.rewardVote
+          (votes :+ currentVote).filter(_ >= 0)
+        }
+
+        val mayBeReward     = lastBlockReward
+        val mayBeTimeToVote = height - activatedAt
+
+        mayBeReward match {
+          case Some(reward) if mayBeTimeToVote > 0 && mayBeTimeToVote % settings.term == 0 =>
+            Some((votes , reward))
+          case None if mayBeTimeToVote == 0 =>
+            Some((Seq(), settings.initial))
+          case _ => None
+        }
+      }
+      .flatMap {
+        case (votes, currentReward) =>
+          val lt = votes.count(_ < currentReward).toDouble / settings.votingInterval
```

Comment:
Please use `Int`s/`Long`s here. No need to convert to `Double`s.  Vote count should just be greater than or equal to `settings.votingInterval / 2 + 1`.